geopandas/tools/geocoding.py

```diff
@@ -111,42 +118,40 @@ def reverse_geocode(points, provider=None, **kwargs):
     1  POINT (-77.03641 38.89766)  1600 Pennsylvania Ave NW, Washington, DC 20006...
     """
 
-    if provider is None:
-        provider = "photon"
-    throttle_time = _get_throttle_time(provider)
-
     return _query(points, False, provider, throttle_time, **kwargs)
 
 
 def _query(data, forward, provider, throttle_time, **kwargs):
     # generic wrapper for calls over lists to geopy Geocoders
-    from geopy.geocoders.base import GeocoderQueryError
+    from geopy.extra.rate_limiter import RateLimiter
     from geopy.geocoders import get_geocoder_for_service
+    from geopy.geocoders.base import GeocoderQueryError
 
-    if forward:
-        if not isinstance(data, pd.Series):
-            data = pd.Series(data)
-    else:
-        if not isinstance(data, geopandas.GeoSeries):
-            data = geopandas.GeoSeries(data)
-
+    if provider is None:
+        provider = "photon"
     if isinstance(provider, str):
         provider = get_geocoder_for_service(provider)
 
+    if throttle_time is None:
+        throttle_time = _get_throttle_time(provider)
+
+    if forward and not isinstance(data, pd.Series):
+        data = pd.Series(data)
+    elif not isinstance(data, geopandas.GeoSeries):
+        data = geopandas.GeoSeries(data)
+
     coder = provider(**kwargs)
+    transform = coder.geocode if forward else coder.reverse
+    transform = RateLimiter(transform, min_delay_seconds=throttle_time)
+
     results = {}
     for i, s in data.items():
         try:
-            if forward:
-                results[i] = coder.geocode(s)
-            else:
-                results[i] = coder.reverse((s.y, s.x), exactly_one=True)
+            results[i] = transform(s) if forward else transform((s.y, s.x))
```

Comment:
You have removed `exactly_one=True` from `reverse` call here. What is the reason?

Reply:
I have checked all geopy providers. Their `geocode` and `reverse` methods all set `exactly_one=True`.
But in case one provider set False in the future. So I will add `exactly_one=True` for `geocode` and `reverse` later.